src/deepsparse/benchmark/api/benchmarker.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import Optional
+
+from deepsparse.benchmark.api.errors import UnclearBenchmarkerModeException
+from deepsparse.benchmark.benchmark_model import benchmark_model
+from deepsparse.benchmark.benchmark_pipeline import benchmark_pipeline
+
+
+class Benchmarker:
+    """
+    Benchmark API
+
+    Input arg to `model`, `pipeline` should be one of:
+     - SparseZoo stub
+     - path to a model.onnx
+     - path to a local folder containing a model.onnx
+     - path to onnx.ModelProto
+
+    Provide the stub/path to one of
+     - model to run deesparse.benchmark
+     - pipeline to run deepsparse.benchmark_pipeline
+    """
+
+    def __init__(
+        self,
+        model: Optional[str] = None,
+        pipeline: Optional[str] = None,
+    ):
+        self._validate_exactly_one_mode_selected(model, pipeline)
+        self.model = model
+        self.pipeline = pipeline
+
+    def __call__(self, **kwargs):
+        if self.model:
+            return benchmark_model(model_path=self.model, **kwargs)
+
+        if self.pipeline:
+            return benchmark_pipeline(model_path=self.pipeline, **kwargs)
+
+    @staticmethod
```

Comment:
can we add a top level `benchmark` function as well so users can just import that if they want to (instead of `Benchmarker.benchmark`